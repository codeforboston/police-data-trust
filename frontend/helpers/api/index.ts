export * from "./base"
export * from "./config"
export * from "./auth"
export * from "./incidents"
export { useMockServiceWorker, apiMode } from "./mocks/browser.setup"
