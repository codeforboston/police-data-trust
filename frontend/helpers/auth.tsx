import { useRouter } from "next/router"
import React, { createContext, useCallback, useContext, useEffect, useMemo, useState } from "react"
import * as api from "./api"
import * as storage from "./storage"

const STORAGE_KEYS = {
  ACCESS_TOKEN: "__pdt_access_token__",
  USER: "__pdt_user__"
}

interface AuthState {
  accessToken?: api.AccessToken
  user?: api.User
  login: (credentials: api.LoginCredentials) => Promise<void>
  logout: () => Promise<void>
  register: (newUser: api.NewUser) => Promise<void>
}

export const AuthContext = createContext<AuthState>(undefined)

export function AuthProvider({ children }: { children: React.ReactNode }) {
  const auth = useAuthenticationHook()
  return <AuthContext.Provider value={auth}>{children}</AuthContext.Provider>
}

export function useAuth() {
  return useContext(AuthContext)
}

/**
 * Renders the given component if authenticated, otherwise redirects to login.
 */
export function requireAuth(Component: () => JSX.Element) {
  return function ProtectedRoute() {
    const { user } = useAuth()
    const router = useRouter()
    const login = "/login"

    useEffect(() => {
      if (!user && router.pathname !== login) {
        router.push(login)
      }
    })

    return user ? <Component /> : null
  }
}

/** Redirects to the given route once the user is authenticated */
export function useRedirectOnAuth(route: string) {
  const { user } = useAuth()
  const router = useRouter()

  useEffect(() => {
    if (user && router.pathname !== route) {
      router.push(route)
    }
  })
}

function useAuthenticationHook(): AuthState {
  const {
    value: accessToken,
    set: setToken,
    clear: clearToken
  } = usePersistence<api.AccessToken>(STORAGE_KEYS.ACCESS_TOKEN)

  const {
    value: user,
    set: setUser,
    clear: clearUser
  } = usePersistence<api.User>(STORAGE_KEYS.USER)

  const login = useCallback(
    async (credentials: api.LoginCredentials) => {
      const accessToken = await api.login(credentials)
      const user = await api.whoami({ accessToken })
      setToken(accessToken)
      setUser(user)
    },
    [setToken, setUser]
  )

  const register = useCallback(
    async (newUser: api.NewUser) => {
      const accessToken = await api.register(newUser)
      const user = await api.whoami({ accessToken })
      setToken(accessToken)
      setUser(user)
    },
    [setToken, setUser]
  )

  const logout = useCallback(async () => {
    clearToken()
    clearUser()
  }, [clearToken, clearUser])

  return useMemo(
    () => ({ accessToken, user, login, register, logout }),
    [accessToken, login, logout, register, user]
  )
}

function usePersistence<T>(storageKey: string) {
  const [value, setValueState] = useState<T>(() => storage.getItem(storageKey))
  const set = useCallback(
      (v: T) => {
        storage.setItem(storageKey, v)
        setValueState(v)
      },
      [storageKey]
    ),
    clear = useCallback(() => {
      storage.removeItem(storageKey)
      setValueState(null)
    }, [storageKey])
  return {
    value,
    set,
    clear
  }
}

/**
 * Seeds auth values for test environments.
 */
export function setAuthForTest(
  user: api.User = {
    active: true,
    email: "testemail@example.com",
    firstname: "FirstTest",
    lastname: "LastTest",
    role: "Public"
  },
  accessToken: api.AccessToken = "faketoken"
) {
  storage.setItem(STORAGE_KEYS.USER, user)
  storage.setItem(STORAGE_KEYS.ACCESS_TOKEN, accessToken)
}

export function clearAuthForTest() {
  storage.removeItem(STORAGE_KEYS.USER)
  storage.removeItem(STORAGE_KEYS.ACCESS_TOKEN)
}
