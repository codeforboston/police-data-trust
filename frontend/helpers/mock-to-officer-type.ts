import { OfficerRecordType, EmploymentType } from "../models/officer"
import { PerpetratorRecordType } from "../models/perpetrator"
import { Incident, Officer, UseOfForce } from "../helpers/api"
import officers from "../models/mock-data/officer.json"
import { IncidentRecordType } from "../models/incident"

export function getOfficerFromMockData(officerId: number) {
  if (officerId >= 0 && officerId < 100) {
    return mockToOfficerType(officers[officerId])
  }
}

export function getPerpetratorFromMockData(perpetratorId: number) {
  if (perpetratorId >= 0 && perpetratorId < 100) {
    return mockToPerpetratorType(officers[perpetratorId])
  }
}

export function mockToOfficerType(officer: typeof officers[0]): OfficerRecordType {
  function mockToWorkHistoryType(workHistory: typeof officer.workHistory): EmploymentType[] {
    const converted: EmploymentType[] = workHistory.map((item) => {
      return {
        agency: {
          agencyName: item.deptName,
          agencyImage: item.deptImage.replace("./frontend/models/mock-data/dept-images", ""),
          agencyHqAddress: item.deptAddress,
          websiteUrl: "https://www.google.com/search?q=police+department"
        },
        currentlyEmployed: true,
        earliestEmployment: new Date(item.dates.split("-")[0].trim()),
        latestEmployment: new Date(item.dates.split("-")[1].trim())
      }
    })
    return converted
  }

  return {
    recordId: officer.id,
    firstName: officer.firstName,
    lastName: officer.lastName,
    dateOfBirth: new Date(officer.birthDate),
    gender: officer.gender,
    race: officer.race,
    badgeNo: officer.badgeNo,
    status: officer.status,
    department: officer.department,
    ethnicity: officer.ethnicity,
    affiliations: officer.affiliations,
    workHistory: mockToWorkHistoryType(officer.workHistory),
    incidents: mockToIncidentType(officer.incidents)
  }
}

export function mockToPerpetratorType(officer: typeof officers[0]): PerpetratorRecordType {
  return {
    recordId: officer.id,
    firstName: officer.firstName,
    lastName: officer.lastName,
    gender: officer.gender,
    race: officer.race
  }
}

export function mockToIncidentType(data: IncidentRecordType[]): IncidentRecordType[] {
  return data
}
