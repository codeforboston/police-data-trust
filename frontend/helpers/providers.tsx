import { QueryClient, QueryClientProvider } from "@tanstack/react-query"
import { ReactQueryDevtools } from "@tanstack/react-query-devtools"
import { Toaster } from "react-hot-toast"
import { AuthProvider } from "./auth"
import { SearchProvider } from "./search"

/**
 * Query Client responsible for housing cache of
 * network requests, serves as basis
 * for React Query functionality
 */
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: 0,
      retry: false
    }
  }
})

/**
 * Wraps components in application providers, which set up contexts to provide
 * services to components.
 */
export const Providers: React.FC = ({ children }) => (
  <>
    <QueryClientProvider client={queryClient}>
      <ReactQueryDevtools />
      <Toaster position="bottom-right" />
      <AuthProvider>
        <SearchProvider>{children}</SearchProvider>
      </AuthProvider>
    </QueryClientProvider>
  </>
)
