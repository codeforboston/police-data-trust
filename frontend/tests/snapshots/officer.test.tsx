import { render } from "../test-utils"
import OfficerView from "../../compositions/officer-view"
import { getOfficerFromMockData } from "../../helpers/mock-to-officer-type"
jest.spyOn(global.Math, "random").mockImplementation(() => 0)
describe("Officer Page", () => {
  it("Renders the officer page correctly", () => {
    const tree = render(<OfficerView {...getOfficerFromMockData(0)} />)
    expect(tree).toMatchSnapshot()
  })
})
