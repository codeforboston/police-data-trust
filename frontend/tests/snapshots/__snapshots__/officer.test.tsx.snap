// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Officer Page Renders the officer page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
      />
      <div
        class="wrapper"
      >
        <svg
          aria-labelledby="svg-inline--fa-title-000000000000"
          class="svg-inline--fa fa-arrow-left fa-w-14 fa-lg backButton"
          data-icon="arrow-left"
          data-prefix="fas"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-000000000000"
          >
            Back
          </title>
          <path
            d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
            fill="currentColor"
          />
        </svg>
        <div
          class="officerView"
        >
          <img
            alt="Profile Picture"
            class="profilePicture"
            src="https://t3.ftcdn.net/jpg/05/16/27/58/360_F_516275801_f3Fsp17x6HQK0xQgDQEELoTuERO4SsWV.jpg"
          />
          <div>
            <div>
              <p
                class="title"
              >
                Officer Record
              </p>
              <p
                class="name"
              >
                Timothy
                 
                Young
              </p>
            </div>
            <div
              class="viewWrapper"
            >
              <div
                class="data"
              >
                <p
                  class="category"
                >
                  Badge Number
                </p>
                <p>
                  30420452
                </p>
              </div>
              <div
                class="data"
              >
                <p
                  class="category"
                >
                  Officer Status
                </p>
                <p>
                  Chief
                </p>
              </div>
              <div
                class="data"
              >
                <p
                  class="category"
                >
                  Department
                </p>
                <p>
                  Houston Police Department
                </p>
              </div>
            </div>
            <div
              class="viewWrapper"
            >
              <div
                class="data"
              >
                <p
                  class="category"
                >
                  Date of Birth
                </p>
                <p>
                  12/15/1976
                </p>
              </div>
              <div
                class="data"
              >
                <p
                  class="category"
                >
                  Gender
                </p>
                <p>
                  Nonbinary
                </p>
              </div>
              <div
                class="data"
              >
                <p
                  class="category"
                >
                  Race
                </p>
                <p>
                  Black or African American
                </p>
              </div>
            </div>
            <div>
              <div
                class="viewWrapper"
              >
                <div
                  class="data"
                >
                  <p
                    class="category"
                  >
                    Ethnicity
                  </p>
                  <p>
                    Hispanic or Latino
                  </p>
                </div>
              </div>
            </div>
            <div
              class="wrapper"
            >
              <p
                class="title"
              >
                Work History:
              </p>
              <div>
                <div
                  class="wrapper"
                >
                  <img
                    alt="San Jose Police Department Patch"
                    class="patch"
                    src="/Boston_Police_Patch.jpg"
                  />
                  <div
                    class="content"
                  >
                    <div
                      class="titleAndDate"
                    >
                      <span
                        class="title"
                      >
                        Detective
                      </span>
                      <span>
                        2014
                         - 
                        2001
                      </span>
                    </div>
                    <a
                      class="link"
                      href="https://www.google.com/search?q=police+department"
                    >
                      San Jose Police Department
                    </a>
                    <p
                      class="address"
                    >
                      (123) 456-7890 * 
                      1191 Elm Ave, Houston, KY 93452
                    </p>
                  </div>
                </div>
                <div
                  class="wrapper"
                >
                  <img
                    alt="Pheonix Police Department Patch"
                    class="patch"
                    src="/San_Jose_Police_Patch.png"
                  />
                  <div
                    class="content"
                  >
                    <div
                      class="titleAndDate"
                    >
                      <span
                        class="title"
                      >
                        Detective
                      </span>
                      <span>
                        1999
                         - 
                        2007
                      </span>
                    </div>
                    <a
                      class="link"
                      href="https://www.google.com/search?q=police+department"
                    >
                      Pheonix Police Department
                    </a>
                    <p
                      class="address"
                    >
                      (123) 456-7890 * 
                      10435 Lemon Ln, San Diego, MN 58574
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <table
                aria-label="Data Table"
                class="dataTable"
                role="table"
              >
                <thead
                  class="dataHeader"
                >
                  <tr
                    role="row"
                  >
                    <th
                      colspan="1"
                      role="columnheader"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                    >
                      Officers involved
                      <span
                        class="sortArrow"
                      >
                          
                      </span>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                    >
                      Incident Type
                      <span
                        class="sortArrow"
                      >
                          
                      </span>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                    >
                      Use of Force
                      <span
                        class="sortArrow"
                      >
                          
                      </span>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                    >
                      Source
                      <span
                        class="sortArrow"
                      >
                          
                      </span>
                    </th>
                    <th
                      colspan="1"
                      role="columnheader"
                    >
                      View Record
                      <span
                        class="sortArrow"
                      >
                          
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  role="rowgroup"
                >
                  <tr
                    class="dataRows"
                    role="row"
                  >
                    <td
                      role="cell"
                    >
                      F.Keaves
                    </td>
                    <td
                      role="cell"
                    >
                      Robbery
                    </td>
                    <td
                      role="cell"
                    >
                      Armed presence
                    </td>
                    <td
                      role="cell"
                    >
                      Police report
                    </td>
                    <td
                      role="cell"
                    >
                      <svg
                        aria-labelledby="svg-inline--fa-title-000000000000"
                        class="svg-inline--fa fa-greater-than fa-w-12 actionBtn"
                        data-icon="greater-than"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 384 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title
                          id="svg-inline--fa-title-000000000000"
                        >
                          View Record
                        </title>
                        <path
                          d="M365.52 209.85L59.22 67.01c-16.06-7.49-35.15-.54-42.64 15.52L3.01 111.61c-7.49 16.06-.54 35.15 15.52 42.64L236.96 256.1 18.49 357.99C2.47 365.46-4.46 384.5 3.01 400.52l13.52 29C24 445.54 43.04 452.47 59.06 445l306.47-142.91a32.003 32.003 0 0 0 18.48-29v-34.23c-.01-12.45-7.21-23.76-18.49-29.01z"
                          fill="currentColor"
                        />
                      </svg>
                    </td>
                  </tr>
                </tbody>
              </table>
              <div
                class="dataFooter"
              >
                <span
                  class="recordCount"
                >
                  Number of results: 
                  1
                </span>
                <span>
                  <span
                    class="pageBtn"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-angle-left fa-w-8 "
                      data-icon="angle-left"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 256 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                  <span
                    class="pageCnt"
                  >
                    1
                     of 
                    1
                  </span>
                  <span
                    class="pageBtn"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-angle-right fa-w-8 "
                      data-icon="angle-right"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 256 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      class="wrapper"
    >
      <svg
        aria-labelledby="svg-inline--fa-title-000000000000"
        class="svg-inline--fa fa-arrow-left fa-w-14 fa-lg backButton"
        data-icon="arrow-left"
        data-prefix="fas"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="svg-inline--fa-title-000000000000"
        >
          Back
        </title>
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
        />
      </svg>
      <div
        class="officerView"
      >
        <img
          alt="Profile Picture"
          class="profilePicture"
          src="https://t3.ftcdn.net/jpg/05/16/27/58/360_F_516275801_f3Fsp17x6HQK0xQgDQEELoTuERO4SsWV.jpg"
        />
        <div>
          <div>
            <p
              class="title"
            >
              Officer Record
            </p>
            <p
              class="name"
            >
              Timothy
               
              Young
            </p>
          </div>
          <div
            class="viewWrapper"
          >
            <div
              class="data"
            >
              <p
                class="category"
              >
                Badge Number
              </p>
              <p>
                30420452
              </p>
            </div>
            <div
              class="data"
            >
              <p
                class="category"
              >
                Officer Status
              </p>
              <p>
                Chief
              </p>
            </div>
            <div
              class="data"
            >
              <p
                class="category"
              >
                Department
              </p>
              <p>
                Houston Police Department
              </p>
            </div>
          </div>
          <div
            class="viewWrapper"
          >
            <div
              class="data"
            >
              <p
                class="category"
              >
                Date of Birth
              </p>
              <p>
                12/15/1976
              </p>
            </div>
            <div
              class="data"
            >
              <p
                class="category"
              >
                Gender
              </p>
              <p>
                Nonbinary
              </p>
            </div>
            <div
              class="data"
            >
              <p
                class="category"
              >
                Race
              </p>
              <p>
                Black or African American
              </p>
            </div>
          </div>
          <div>
            <div
              class="viewWrapper"
            >
              <div
                class="data"
              >
                <p
                  class="category"
                >
                  Ethnicity
                </p>
                <p>
                  Hispanic or Latino
                </p>
              </div>
            </div>
          </div>
          <div
            class="wrapper"
          >
            <p
              class="title"
            >
              Work History:
            </p>
            <div>
              <div
                class="wrapper"
              >
                <img
                  alt="San Jose Police Department Patch"
                  class="patch"
                  src="/Boston_Police_Patch.jpg"
                />
                <div
                  class="content"
                >
                  <div
                    class="titleAndDate"
                  >
                    <span
                      class="title"
                    >
                      Detective
                    </span>
                    <span>
                      2014
                       - 
                      2001
                    </span>
                  </div>
                  <a
                    class="link"
                    href="https://www.google.com/search?q=police+department"
                  >
                    San Jose Police Department
                  </a>
                  <p
                    class="address"
                  >
                    (123) 456-7890 * 
                    1191 Elm Ave, Houston, KY 93452
                  </p>
                </div>
              </div>
              <div
                class="wrapper"
              >
                <img
                  alt="Pheonix Police Department Patch"
                  class="patch"
                  src="/San_Jose_Police_Patch.png"
                />
                <div
                  class="content"
                >
                  <div
                    class="titleAndDate"
                  >
                    <span
                      class="title"
                    >
                      Detective
                    </span>
                    <span>
                      1999
                       - 
                      2007
                    </span>
                  </div>
                  <a
                    class="link"
                    href="https://www.google.com/search?q=police+department"
                  >
                    Pheonix Police Department
                  </a>
                  <p
                    class="address"
                  >
                    (123) 456-7890 * 
                    10435 Lemon Ln, San Diego, MN 58574
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div>
            <table
              aria-label="Data Table"
              class="dataTable"
              role="table"
            >
              <thead
                class="dataHeader"
              >
                <tr
                  role="row"
                >
                  <th
                    colspan="1"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    Officers involved
                    <span
                      class="sortArrow"
                    >
                        
                    </span>
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    Incident Type
                    <span
                      class="sortArrow"
                    >
                        
                    </span>
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    Use of Force
                    <span
                      class="sortArrow"
                    >
                        
                    </span>
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    Source
                    <span
                      class="sortArrow"
                    >
                        
                    </span>
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    View Record
                    <span
                      class="sortArrow"
                    >
                        
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  class="dataRows"
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    F.Keaves
                  </td>
                  <td
                    role="cell"
                  >
                    Robbery
                  </td>
                  <td
                    role="cell"
                  >
                    Armed presence
                  </td>
                  <td
                    role="cell"
                  >
                    Police report
                  </td>
                  <td
                    role="cell"
                  >
                    <svg
                      aria-labelledby="svg-inline--fa-title-000000000000"
                      class="svg-inline--fa fa-greater-than fa-w-12 actionBtn"
                      data-icon="greater-than"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 384 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <title
                        id="svg-inline--fa-title-000000000000"
                      >
                        View Record
                      </title>
                      <path
                        d="M365.52 209.85L59.22 67.01c-16.06-7.49-35.15-.54-42.64 15.52L3.01 111.61c-7.49 16.06-.54 35.15 15.52 42.64L236.96 256.1 18.49 357.99C2.47 365.46-4.46 384.5 3.01 400.52l13.52 29C24 445.54 43.04 452.47 59.06 445l306.47-142.91a32.003 32.003 0 0 0 18.48-29v-34.23c-.01-12.45-7.21-23.76-18.49-29.01z"
                        fill="currentColor"
                      />
                    </svg>
                  </td>
                </tr>
              </tbody>
            </table>
            <div
              class="dataFooter"
            >
              <span
                class="recordCount"
              >
                Number of results: 
                1
              </span>
              <span>
                <span
                  class="pageBtn"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-angle-left fa-w-8 "
                    data-icon="angle-left"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 256 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
                <span
                  class="pageCnt"
                >
                  1
                   of 
                  1
                </span>
                <span
                  class="pageBtn"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-angle-right fa-w-8 "
                    data-icon="angle-right"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 256 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
